File: mamba-agent/src/main/java/com/mamba/agent/data/message/ChatMessage.java
package com.mamba.agent.data.message;

import com.mamba.agent.model.chat.ChatModel;
import com.mamba.agent.model.chat.StreamingChatModel;

/**
 * Represents a chat message.
 * Used together with {@link ChatModel} and {@link StreamingChatModel}.
 *
 * @see SystemMessage
 * @see UserMessage
 * @see AiMessage
 * @see ToolExecutionResultMessage
 * @see CustomMessage
 */
public interface ChatMessage {

    /**
     * The type of the message.
     *
     * @return the type of the message
     */
    ChatMessageType type();
}

File: agent-core/src/main/java/com/mamba/picme/agent/core/api/ChatMessage.kt
package com.mamba.picme.agent.core.api

sealed interface ChatMessage

data class SystemMessage(val text: String) : ChatMessage
data class UserMessage(val text: String) : ChatMessage
data class AiMessage(
    val text: String,
    val toolExecutionRequests: List<ToolExecutionRequest> = emptyList()
) : ChatMessage

data class ToolExecutionResultMessage(
    val toolExecutionRequest: ToolExecutionRequest,
    val text: String
) : ChatMessage
